### ISSUE TITLE
Invalid token handling / CloudFlare ratelimits

### ISSUE THREAD

**Issue by napstr** [Discord4J/Discord4J#917]
I've recently ran into issues of my token getting revoked (still investigating the root cause for that, but that's not what this ticket is about).

When continuing to hit the Discord API with an invalid token, eventually the bot will be IP banned by Cloudflare for one 1 hour or 24 hours.
I'm getting the impression that Discord4J could be handling this better. I've got my token revoked a few times with JDA, but only once had a CF ban, and only for one hour. With D4J, I'm at a 24 hour ban pretty shortly (about 8 minutes) after a token reset.

**To Reproduce:** 
idk, use up all ur sessions on a 500 shard bot? I dont recommend this lol
maybe starting with an invalid token works as well

**Expected Behavior:** 
When encountering 401s, its probably a good idea to havea default behaviour in place that severly throttles further requests to Discord.

**Actual Behavior:** 

**Version:** 
3.2.0-SNAPSHOT

**Other:**

**Comment by quanticc** [comment]
I'd appreciate suggesting any other way to reproduce this. So far token reset from developer portal triggers a reconnect and then bot is disconnected, so works as intended (GatewayDiscordClient). Have two questions:

How does your client builder looks like? Wondering if there's a special or custom ShardingStrategy

Are you using a separate RestClient for executing requests, or otherwise run requests outside of the bot event loop?

**Comment by napstr** [comment]
Investigating my metrics, they indicate that the bot continued to try to send messages at a rather high rate even after the token was invalidated. Unfortunately I dont see what exactly was doing the sending at that high frequency in my logs - maybe a badly configured `retry` somewhere - in any case I think I can rule out D4J as the root cause.

**Comment by quanticc** [comment]
Closing for now, if there's a way to repro an issue with D4J we can reopen. If there's a feature request like implementing a killswitch Router, let's track it on a separate issue.
